core/tests/window/test_window.py

```diff
@@ -279,14 +280,99 @@ def test_visibility(window, app):
 def test_full_screen(window, app):
     """A window can be set full screen."""
     assert not window.full_screen
-
     window.full_screen = True
     assert window.full_screen
-    assert_action_performed_with(window, "set full screen", full_screen=True)
+    assert_action_performed_with(
+        window,
+        "set window state to WindowState.FULLSCREEN",
+        state=WindowState.FULLSCREEN,
+    )
 
     window.full_screen = False
     assert not window.full_screen
-    assert_action_performed_with(window, "set full screen", full_screen=False)
+    assert_action_performed_with(
+        window,
+        "set window state to WindowState.NORMAL",
+        state=WindowState.NORMAL,
+    )
+
+    # Setting full screen to False when window is not in full screen, is a no-op
+    #
+    # We cannot assert that the action was not performed, since the same action
+    # was performed previously and would still be in EventLog. Therefore, we
+    # cannot check if the action was done by the first call or the second call.
+    # Hence, this is just to reach coverage.
+    assert not window.full_screen
+    window.full_screen = False
+    # assert_action_not_performed(window, "set window state to WindowState.NORMAL")
+
+
+def test_window_state(window):
+    """A window can have different states."""
+    assert window.state == WindowState.NORMAL
+
+    for state in WindowState:
```

Comment:
For loops in tests are generally an anti pattern - you should be using a parameterised test so that specific cases in the for loop can be identified as a problem. In this case, we're interested in all possible "initial state" to "final state" transitions.

Reply:
I have parameterized the test.